[Feature Request] Return a list of events based at particular point in time

Hello,

It would be nice to have a function that takes a datetime object as input and returns a list of events that are active at specific moment.

As a use case, we have a calendar with blackouts (represented as events). On incoming alert from monitoring system, we should check if there is a blackout on specific server at the moment. If blackout found, log actions and drop alert.

This can be something like this:
```python
    def at_moment(self, now: datetime.datetime = None) -> List[Event]:
        """This function should return the list of events active at the specific point in time"""
        if not now:
            now = datetime.datetime.now()
        # If event starts somewhere previous day with long duration
        start = now - datetime.timedelta(days=1)
        end = now + datetime.timedelta(days=1)

        events = self.between(start.date().timetuple()[:3], end.date().timetuple()[:3])
        return [e for e in events if e.get('dtstart').dt < now < e.get('dtend').dt]
```
I'm not sure what's better way to identify the start and the end for `between` function. If we'll have an event with duration more than one day, it will not work.

Hey,

Thanks for your response. I have done some tests on my side and `between` function works fine with double `datetime` object as input.

For me, the function `between` means that it will return all the events that fall into that defined period/span (as it's stated in docstring). It does not explicitly states, that it can also work in reverse: give me all events that occur at specific `datetime`. And moreover, oddly you need to put same `datetime` twice for `start` and `stop` parameters.

For me, it sould be more explicit. For example putting an alias function like this:
```python
    def at_dt(self, dt: datetime):
        return self.between(dt, dt)
```
Or maybe, allowing `at` function to accept `datetime` as input. As of now, following test fails:
```python
def test_at_for_datetime(calendars):
    now = datetime(2018, 1, 15, 10, 30, 0, 0)
    events = calendars.duration.at(now)
    assert len(events) == 1
    assert events[0].get('summary') == 'Duration Event'
```
Pytest output:
```
self = <recurring_ical_events.UnfoldableCalendar object at 0x000001A8DA6AC748>, date = datetime.datetime(2018, 1, 15, 10, 30)

    def at(self, date):
        """Return all events within the next 24 hours of starting at the given day.

        - date can be a year like (2019,) or 2099
        - a month like (2019, 1) for January of 2019
        - a day like (2019, 1, 1) for the first of January 2019
        - a day with hours, (2019, 1, 1, 1)
        - a day with minutes, (2019, 1, 1, 1, 1)
        - a day with seconds, (2019, 1, 1, 1, 1, 1)
        """
        if isinstance(date, int):
            date = (date,)
        if isinstance(date, str):
            assert len(date) == 8 and date.isdigit(), "format yyyymmdd expected"
            date = (int(date[:4], 10), int(date[4:6], 10), int(date[6:]))
>       if len(date) == 1:
E       TypeError: object of type 'datetime.datetime' has no len()

recurring_ical_events.py:265: TypeError
```

--- a/.gitignore
+++ b/.gitignore
@@ -4,3 +4,5 @@ __pycache__
 *egg-info
 /build
 /dist
+.pytest_cache/
+.vscode/

--- a/recurring_ical_events.py
+++ b/recurring_ical_events.py
@@ -262,6 +262,10 @@ class UnfoldableCalendar:
         if isinstance(date, str):
             assert len(date) == 8 and date.isdigit(), "format yyyymmdd expected"
             date = (int(date[:4], 10), int(date[4:6], 10), int(date[6:]))
+        if isinstance(date, datetime.datetime):
+            return self.between(date, date)
+        if isinstance(date, datetime.date):
+            return self.between(date, date + datetime.timedelta(days=1))
         if len(date) == 1:
             return self.between((date[0], 1, 1), (date[0] + 1, 1, 1))
         if len(date) == 2:
@@ -269,8 +273,8 @@ class UnfoldableCalendar:
             if month == 12:
                 return self.between((year, 12, 1), (year + 1, 1, 1))
             return self.between((year, month, 1), (year, month + 1, 1))
-        datetime = self.to_datetime(date)
-        return self.between(datetime, datetime + self._DELTAS[len(date) - 3])
+        dt = self.to_datetime(date)
+        return self.between(dt, dt + self._DELTAS[len(date) - 3])
     
     def between(self, start, stop): # TODO: add parameters from time_span_contains_event
         """Return events at a time between start (inclusive) and end (inclusive)"""

--- a/test/calendars/duration.ics
+++ b/test/calendars/duration.ics
@@ -5,13 +5,13 @@ BEGIN:VEVENT
 DTSTART:20180110
 DURATION:P3D
 DESCRIPTION:Event with duration (3 days), instead of explicit end.
-SUMMARY:Duration Event
+SUMMARY:Duration Event 1
 END:VEVENT
 BEGIN:VEVENT
 DTSTART:20180115T100000
 DURATION:PT3H
 DESCRIPTION:Event with duration (3 hours), instead of explicit end.
-SUMMARY:Duration Event
+SUMMARY:Duration Event 2
 END:VEVENT
 BEGIN:VEVENT
 DTSTART:20180120T120000

--- a/test/test_at_function.py
+++ b/test/test_at_function.py
@@ -1,0 +1,16 @@
+import pytest
+from datetime import date, datetime
+
+
+@pytest.mark.parametrize("a_date", [
+    # Year
+    2019, (2019,),
+    # Month
+    (2019, 1),
+    # Day
+    (2019, 1, 1), date(2019, 1, 1), "20190101",
+    # Datetime
+    datetime(2019, 1, 1, 10, 0, 0)
+])
+def test_at_input_arguments(a_date, calendars):
+    calendars.duration.at(a_date)

--- a/test/test_at_function.py
+++ b/test/test_at_function.py
@@ -1,16 +1,33 @@
 import pytest
 from datetime import date, datetime
+from icalendar import Event
 
 
-@pytest.mark.parametrize("a_date", [
+@pytest.mark.parametrize("a_date,count", [
     # Year
-    2019, (2019,),
+    (2018, 3), ((2018,), 3),
     # Month
-    (2019, 1),
+    ((2018, 1), 3),
     # Day
-    (2019, 1, 1), date(2019, 1, 1), "20190101",
+    ((2018, 1, 11), 1), (date(2018, 1, 11), 1), ("20180111", 1),
     # Datetime
-    datetime(2019, 1, 1, 10, 0, 0)
+    (datetime(2018, 1, 11, 10, 0, 0), 0)
 ])
-def test_at_input_arguments(a_date, calendars):
-    calendars.duration.at(a_date)
+def test_at_input_arguments(a_date, count, calendars):
+    events = calendars.duration.at(a_date)
+    assert isinstance(events, list)
+    for event in events:
+        assert isinstance(event, Event)
+    assert len(events) == count
+
+
+@pytest.mark.parametrize("a_date,count", [
+    (datetime(2018, 1, 15, 9, 59, 0), 0),
+    (datetime(2018, 1, 15, 10, 0, 0), 1),
+    (datetime(2018, 1, 15, 11, 0, 0), 1),
+    (datetime(2018, 1, 15, 13, 0, 0), 0),
+    (datetime(2018, 1, 15, 13, 1, 0), 0),
+])
+def test_at_datetime_input(a_date, count, calendars):
+    events = calendars.duration.at(a_date)
+    assert len(events) == count

--- a/test/test_at_function.py
+++ b/test/test_at_function.py
@@ -9,8 +9,10 @@ from datetime import date, datetime
     ((2018, 1), 3),
     # Day
     ((2018, 1, 11), 1), (date(2018, 1, 11), 1), ("20180111", 1),
+    ((2018, 1, 9), 0), (date(2018, 1, 9), 0), ("20180109", 0),
     # Datetime
-    (datetime(2018, 1, 11, 10, 0, 0), 0)
+    (datetime(2018, 1, 11, 10, 0, 0), 1),
+    (datetime(2018, 1, 9, 10, 0, 0), 0)
 ])
 def test_at_input_arguments(a_date, count, calendars):
     events = calendars.duration.at(a_date)

--- a/test/test_at_function.py
+++ b/test/test_at_function.py
@@ -1,6 +1,5 @@
 import pytest
 from datetime import date, datetime
-from icalendar import Event
 
 
 @pytest.mark.parametrize("a_date,count", [
@@ -14,20 +13,5 @@ from icalendar import Event
     (datetime(2018, 1, 11, 10, 0, 0), 0)
 ])
 def test_at_input_arguments(a_date, count, calendars):
-    events = calendars.duration.at(a_date)
-    assert isinstance(events, list)
-    for event in events:
-        assert isinstance(event, Event)
-    assert len(events) == count
-
-
-@pytest.mark.parametrize("a_date,count", [
-    (datetime(2018, 1, 15, 9, 59, 0), 0),
-    (datetime(2018, 1, 15, 10, 0, 0), 1),
-    (datetime(2018, 1, 15, 11, 0, 0), 1),
-    (datetime(2018, 1, 15, 13, 0, 0), 0),
-    (datetime(2018, 1, 15, 13, 1, 0), 0),
-])
-def test_at_datetime_input(a_date, count, calendars):
     events = calendars.duration.at(a_date)
     assert len(events) == count

